komodo-relational/src/main/java/org/komodo/relational/workspace/WorkspaceManager.java

```diff
@@ -184,15 +260,16 @@ public KomodoType getTypeIdentifier(UnitOfWork uow) {
      *
      * @param repository remove instance with given repository
      */
-    public static void uncacheInstance(final Repository repository) {
+    public static void uncacheInstance(final Repository repository, final String owner) {
         if (repository == null)
             return;
 
-        instances.remove(repository.getId());
+        instances.remove(new CacheKey(repository.getId(), owner));
     }
 
-    private WorkspaceManager(UnitOfWork uow, Repository repository ) throws KException {
-        super( repository, RepositoryImpl.WORKSPACE_ROOT, 0 );
+    private WorkspaceManager(Repository repository, UnitOfWork uow ) throws KException {
```

Comment:
Other than the repository, this constructor only really needs information about the user. Maybe we should pass in only what is needed. Wdyt?